// react-native-iap/lib/module/types/apple.js
export let ReceiptValidationStatus;

(function (ReceiptValidationStatus) {
  ReceiptValidationStatus[ReceiptValidationStatus["SUCCESS"] = 0] = "SUCCESS";
  ReceiptValidationStatus[ReceiptValidationStatus["INVALID_JSON"] = 21000] = "INVALID_JSON";
  ReceiptValidationStatus[ReceiptValidationStatus["INVALID_RECEIPT_DATA"] = 21002] = "INVALID_RECEIPT_DATA";
  ReceiptValidationStatus[ReceiptValidationStatus["COULT_NOT_AUTHENTICATE"] = 21003] = "COULT_NOT_AUTHENTICATE";
  ReceiptValidationStatus[ReceiptValidationStatus["INVALID_SECRET"] = 21004] = "INVALID_SECRET";
  ReceiptValidationStatus[ReceiptValidationStatus["UNAVAILABLE"] = 21005] = "UNAVAILABLE";
  ReceiptValidationStatus[ReceiptValidationStatus["EXPIRED_SUBSCRIPTION"] = 21006] = "EXPIRED_SUBSCRIPTION";
  ReceiptValidationStatus[ReceiptValidationStatus["TEST_RECEIPT"] = 21007] = "TEST_RECEIPT";
  ReceiptValidationStatus[ReceiptValidationStatus["PROD_RECEIPT"] = 21008] = "PROD_RECEIPT";
  ReceiptValidationStatus[ReceiptValidationStatus["COULD_NOT_AUTHORIZE"] = 21010] = "COULD_NOT_AUTHORIZE";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_00"] = 21100] = "INTERNAL_ERROR_00";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_01"] = 21101] = "INTERNAL_ERROR_01";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_02"] = 21102] = "INTERNAL_ERROR_02";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_03"] = 21103] = "INTERNAL_ERROR_03";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_04"] = 21104] = "INTERNAL_ERROR_04";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_05"] = 21105] = "INTERNAL_ERROR_05";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_06"] = 21106] = "INTERNAL_ERROR_06";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_07"] = 21107] = "INTERNAL_ERROR_07";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_08"] = 21108] = "INTERNAL_ERROR_08";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_09"] = 21109] = "INTERNAL_ERROR_09";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_10"] = 21110] = "INTERNAL_ERROR_10";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_11"] = 21111] = "INTERNAL_ERROR_11";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_12"] = 21112] = "INTERNAL_ERROR_12";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_13"] = 21113] = "INTERNAL_ERROR_13";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_14"] = 21114] = "INTERNAL_ERROR_14";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_15"] = 21115] = "INTERNAL_ERROR_15";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_16"] = 21116] = "INTERNAL_ERROR_16";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_17"] = 21117] = "INTERNAL_ERROR_17";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_18"] = 21118] = "INTERNAL_ERROR_18";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_19"] = 21119] = "INTERNAL_ERROR_19";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_20"] = 21120] = "INTERNAL_ERROR_20";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_21"] = 21121] = "INTERNAL_ERROR_21";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_22"] = 21122] = "INTERNAL_ERROR_22";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_23"] = 21123] = "INTERNAL_ERROR_23";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_24"] = 21124] = "INTERNAL_ERROR_24";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_25"] = 21125] = "INTERNAL_ERROR_25";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_26"] = 21126] = "INTERNAL_ERROR_26";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_27"] = 21127] = "INTERNAL_ERROR_27";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_28"] = 21128] = "INTERNAL_ERROR_28";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_29"] = 21129] = "INTERNAL_ERROR_29";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_30"] = 21130] = "INTERNAL_ERROR_30";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_31"] = 21131] = "INTERNAL_ERROR_31";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_32"] = 21132] = "INTERNAL_ERROR_32";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_33"] = 21133] = "INTERNAL_ERROR_33";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_34"] = 21134] = "INTERNAL_ERROR_34";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_35"] = 21135] = "INTERNAL_ERROR_35";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_36"] = 21136] = "INTERNAL_ERROR_36";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_37"] = 21137] = "INTERNAL_ERROR_37";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_38"] = 21138] = "INTERNAL_ERROR_38";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_39"] = 21139] = "INTERNAL_ERROR_39";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_40"] = 21140] = "INTERNAL_ERROR_40";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_41"] = 21141] = "INTERNAL_ERROR_41";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_42"] = 21142] = "INTERNAL_ERROR_42";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_43"] = 21143] = "INTERNAL_ERROR_43";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_44"] = 21144] = "INTERNAL_ERROR_44";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_45"] = 21145] = "INTERNAL_ERROR_45";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_46"] = 21146] = "INTERNAL_ERROR_46";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_47"] = 21147] = "INTERNAL_ERROR_47";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_48"] = 21148] = "INTERNAL_ERROR_48";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_49"] = 21149] = "INTERNAL_ERROR_49";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_50"] = 21150] = "INTERNAL_ERROR_50";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_51"] = 21151] = "INTERNAL_ERROR_51";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_52"] = 21152] = "INTERNAL_ERROR_52";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_53"] = 21153] = "INTERNAL_ERROR_53";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_54"] = 21154] = "INTERNAL_ERROR_54";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_55"] = 21155] = "INTERNAL_ERROR_55";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_56"] = 21156] = "INTERNAL_ERROR_56";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_57"] = 21157] = "INTERNAL_ERROR_57";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_58"] = 21158] = "INTERNAL_ERROR_58";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_59"] = 21159] = "INTERNAL_ERROR_59";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_60"] = 21160] = "INTERNAL_ERROR_60";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_61"] = 21161] = "INTERNAL_ERROR_61";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_62"] = 21162] = "INTERNAL_ERROR_62";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_63"] = 21163] = "INTERNAL_ERROR_63";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_64"] = 21164] = "INTERNAL_ERROR_64";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_65"] = 21165] = "INTERNAL_ERROR_65";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_66"] = 21166] = "INTERNAL_ERROR_66";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_67"] = 21167] = "INTERNAL_ERROR_67";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_68"] = 21168] = "INTERNAL_ERROR_68";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_69"] = 21169] = "INTERNAL_ERROR_69";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_70"] = 21170] = "INTERNAL_ERROR_70";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_71"] = 21171] = "INTERNAL_ERROR_71";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_72"] = 21172] = "INTERNAL_ERROR_72";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_73"] = 21173] = "INTERNAL_ERROR_73";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_74"] = 21174] = "INTERNAL_ERROR_74";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_75"] = 21175] = "INTERNAL_ERROR_75";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_76"] = 21176] = "INTERNAL_ERROR_76";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_77"] = 21177] = "INTERNAL_ERROR_77";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_78"] = 21178] = "INTERNAL_ERROR_78";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_79"] = 21179] = "INTERNAL_ERROR_79";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_80"] = 21180] = "INTERNAL_ERROR_80";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_81"] = 21181] = "INTERNAL_ERROR_81";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_82"] = 21182] = "INTERNAL_ERROR_82";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_83"] = 21183] = "INTERNAL_ERROR_83";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_84"] = 21184] = "INTERNAL_ERROR_84";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_85"] = 21185] = "INTERNAL_ERROR_85";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_86"] = 21186] = "INTERNAL_ERROR_86";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_87"] = 21187] = "INTERNAL_ERROR_87";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_88"] = 21188] = "INTERNAL_ERROR_88";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_89"] = 21189] = "INTERNAL_ERROR_89";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_90"] = 21190] = "INTERNAL_ERROR_90";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_91"] = 21191] = "INTERNAL_ERROR_91";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_92"] = 21192] = "INTERNAL_ERROR_92";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_93"] = 21193] = "INTERNAL_ERROR_93";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_94"] = 21194] = "INTERNAL_ERROR_94";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_95"] = 21195] = "INTERNAL_ERROR_95";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_96"] = 21196] = "INTERNAL_ERROR_96";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_97"] = 21197] = "INTERNAL_ERROR_97";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_98"] = 21198] = "INTERNAL_ERROR_98";
  ReceiptValidationStatus[ReceiptValidationStatus["INTERNAL_ERROR_99"] = 21199] = "INTERNAL_ERROR_99";
})(ReceiptValidationStatus || (ReceiptValidationStatus = {}));

export let SubscriptionExpirationIntent;

(function (SubscriptionExpirationIntent) {
  SubscriptionExpirationIntent["CUSTOMER_CANCELED"] = "1";
  SubscriptionExpirationIntent["BILLING_ERROR"] = "2";
  SubscriptionExpirationIntent["DENIED_PRICE_INCREASE"] = "3";
  SubscriptionExpirationIntent["PRODUCT_NOT_AVAILABLE"] = "4";
  SubscriptionExpirationIntent["UNKNOWN_ERROR"] = "5";
})(SubscriptionExpirationIntent || (SubscriptionExpirationIntent = {}));

export let SubscriptionRetryFlag;

(function (SubscriptionRetryFlag) {
  SubscriptionRetryFlag["ACTIVE"] = "1";
  SubscriptionRetryFlag["STOPPED"] = "0";
})(SubscriptionRetryFlag || (SubscriptionRetryFlag = {}));

export let CancellationReason;

(function (CancellationReason) {
  CancellationReason["ACTUAL_ISSUE"] = "1";
  CancellationReason["OTHER_REASON"] = "0";
})(CancellationReason || (CancellationReason = {}));

export let SubscriptionAutoRenewStatus;

(function (SubscriptionAutoRenewStatus) {
  SubscriptionAutoRenewStatus["ACTIVE"] = "1";
  SubscriptionAutoRenewStatus["STOPPED"] = "0";
})(SubscriptionAutoRenewStatus || (SubscriptionAutoRenewStatus = {}));

export let SubscriptionPriceConsentStatus;

(function (SubscriptionPriceConsentStatus) {
  SubscriptionPriceConsentStatus["AGREED"] = "1";
  SubscriptionPriceConsentStatus["NO_ACTION"] = "0";
})(SubscriptionPriceConsentStatus || (SubscriptionPriceConsentStatus = {}));
